x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/InferenceIngestIT.java

```diff
@@ -135,13 +144,23 @@ public void testPathologicalPipelineCreationAndDeletion() throws Exception {
             try {
                 Response statsResponse = client().performRequest(new Request("GET",
                     "_ml/trained_models/" + classificationModelId + "/_stats"));
-                String response = EntityUtils.toString(statsResponse.getEntity());
-                assertThat(response, containsString("\"inference_count\":10"));
-                assertThat(response, containsString("\"cache_miss_count\":30"));
+                try (XContentParser parser = createParser(JsonXContent.jsonXContent, statsResponse.getEntity().getContent())) {
```

Comment:
If I see correctly, the pattern of parsing the response and verifying `getInferenceCount` is repeated 4 times.
Please consider extracting the snippet to a separate method parameterized by the expected inference count.